Scrapers/VenueExplorer.py

```diff
@@ -1 +1,130 @@
+
+    # extract general capacity
+    capacity_html = indiv_venue.find_all('div', class_='col-lg-3 col-md-3 col-sm-3 text-center')[2]
+    capacity = ''
+    if capacity_html:
+        capacity = capacity_html.text.split(' ')[0].strip()
+
+    # extract the different rooms and their capacity in this venue
+    rooms_html = indiv_venue.find_all('ul', class_='space')
+    rooms = []
+    if rooms_html and len(rooms_html) > 1:
+        for index, room in enumerate(rooms_html[1].find_all('li')):
+            if room:
+                tokens = room.text.split(' ')
+                room_name = tokens[3].strip()
+                room_cap = tokens[0].strip()
+                rooms.append({
+                    "room" : room_name,
+                    "capacity" : room_cap
+                })
+
+    # for room_html in rooms_html.find_all('li'):
+    #     if room_html:
+    #         texts = room_html.text.split(' ')
+    #         capacity = texts[0].strip() # but this one sometimes is empty. might need to take the above capacity and take it
+    #     # as a general capacity across all events
+    #         roomName = texts[3].strip()
+    #         roomNames.append(roomName)
+    #     else:
+    #         capacity = indiv_venue.find_all('div', class_='col-lg-3 col-md-3 col-sm-3 text-center')[2].text.split(' ')[0].strip()
+    #         break
+
+    # extract location
+    location_html = indiv_venue.find_all('div', class_='col-lg-3 col-md-3 col-sm-3 text-center')[3]
+    location = ''
+    if location_html:
+        location = location_html.text.strip()
+
+    # extract facilities
+    facs = indiv_venue.find('ul', class_='feature')
+    facilities = []
+    for fac in facs.find_all('li'):
+        facilities.append(fac.text.strip())
+
+    # extract description
+    desc_html = indiv_venue.find('div', class_='quick-enquiry').p
+    desc = ''
+    if desc_html:
+        desc = desc_html.text.strip()
+    
+    # no ratings, promos
+    ratings = ''
+    promos = ''
+
+    # create Venue Class with all inputs
+    singleVenue = VenueClass.Venue(
+        ratings, venue_link, images, title, location, tags, price, capacity, desc, facilities, rooms, promos)
+
+    # add this venue to the Venue Class array
+    if(not singleVenue in allVenues):
+        allVenues.append(singleVenue.getVenue())
+
+
+##### MAIN FUNCTION #####
+currentPage = 1  # Initialised to first 
+
+while True:
+
+    # Request html from website
+    url = "https://www.venuexplorer.com.sg/all-venue/{}".format(currentPage)
+    response = get(url)
+    html_soup = BeautifulSoup(response.text, 'html.parser')
+
+    # Check for last page
+    venues = html_soup.find_all('div', class_='venue-box')
+    if(len(venues) == 0):
+        break
+
+    # extract venue info for all venues on this page
+    for venue in venues:
+        extractVenue(venue)
+
+    # Increase counter to go next page
+    currentPage += 1
+
+# write all data into a json file
+os.chdir(r'../Data/')
```

Comment:
**with open('Data/VenuExplorer.json', 'w', encoding='utf-8') as outfile:
    json.dump(allVenues,
              outfile, ensure_ascii=False)**



Could write data in the way above (I am gonna change those write code lines to this for the other scrapers i did. The diff is that i add an encoding of utf-8 chars cos i noticed that some chars that were scrapped might be chinese chars or some other weird chars. So in order to preserve them, we can use the below encoding method before saving as json. I copied this from some stackoverflow post lolll ): 

Whenever you run the Scraper, make sure you are in the directory level same as the Scrapers folder.